src/kernels/jupyter/jupyterUtils.ts

```diff
@@ -127,8 +129,11 @@ export function createRemoteConnectionInfo(
         dispose: noop,
         rootDirectory: Uri.file(''),
         workingDirectory: serverUri?.workingDirectory,
-        getAuthHeader: serverUri ? () => getJupyterServerUri(uri)?.authorizationHeader : undefined,
-        getWebsocketProtocols: serverUri ? () => getJupyterServerUri(uri)?.webSocketProtocols || [] : () => [],
+        // For remote jupyter servers that are managed by us, we can provide the auth header.
+        // Its crucial this is set to undefined, else password retrieval will not be attempted.
+        getAuthHeader: serverUri && !serverId.startsWith('_builtin') ? () => serverUri?.authorizationHeader : undefined,
```

Comment:
This resulted in auth never working, even after entering the password it would never authenticate against the server